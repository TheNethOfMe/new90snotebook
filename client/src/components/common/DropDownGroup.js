import React from "react";
import classnames from "classnames";
import PropTypes from "prop-types";

const DropDownGroup = ({
  name,
  label,
  error,
  info,
  onChange,
  disabled,
  selected,
  options
}) => {
  return (
    <div className="input">
      <label className="input__label" htmlFor={name}>
        {label}
      </label>
      <select
        className={classnames("select__field", {
          "input__invalid-input": error
        })}
        name={name}
        onChange={onChange}
        disabled={disabled}
        defaultValue={selected}
      >
        {options.map(opt => {
          return (
            <option key={opt.val} value={opt.val}>
              {opt.display}
            </option>
          );
        })}
      </select>
      {info && <p className="input__info">{info}</p>}
      {error && <p className="input__error">{error}</p>}
    </div>
  );
};

DropDownGroup.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  info: PropTypes.string,
  error: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  disabled: PropTypes.bool.isRequired,
  selected: PropTypes.string,
  options: PropTypes.array.isRequired
};

DropDownGroup.defaultProps = {
  type: "text",
  disabled: false
};

export default DropDownGroup;
