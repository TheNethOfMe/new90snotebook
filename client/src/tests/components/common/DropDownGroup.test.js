import React from "react";
import { shallow } from "enzyme";
import DropDownGroup from "../../../components/common/DropDownGroup";

let wrapper, onChange, error, info;

beforeEach(() => {
  onChange = jest.fn();
  wrapper = shallow(
    <DropDownGroup
      name="email"
      label="Email"
      error={error}
      onChange={onChange}
      info={info}
      options={[
        {
          val: "one",
          display: "Option One"
        },
        {
          val: "two",
          display: "Option Two"
        }
      ]}
    />
  );
});

test("should render the select drop-down", () => {
  expect(wrapper).toMatchSnapshot();
});

test("should fire onChange when option is selected", () => {
  wrapper
    .find("select")
    .at(0)
    .simulate("change", "two");
  expect(onChange).toHaveBeenCalledWith("two");
});
