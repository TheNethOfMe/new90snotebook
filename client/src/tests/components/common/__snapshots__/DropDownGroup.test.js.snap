// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the select drop-down 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DropDownGroup
    disabled={false}
    label="Email"
    name="email"
    onChange={[MockFunction]}
    options={
      Array [
        Object {
          "display": "Option One",
          "val": "one",
        },
        Object {
          "display": "Option Two",
          "val": "two",
        },
      ]
    }
    type="text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          className="input__label"
          htmlFor="email"
        >
          Email
        </label>,
        <select
          className="select__field"
          disabled={false}
          name="email"
          onChange={[MockFunction]}
        >
          <option
            value="one"
          >
            Option One
          </option>
          <option
            value="two"
          >
            Option Two
          </option>
        </select>,
        undefined,
        undefined,
      ],
      "className": "input",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Email",
          "className": "input__label",
          "htmlFor": "email",
        },
        "ref": null,
        "rendered": "Email",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value="one"
            >
              Option One
            </option>,
            <option
              value="two"
            >
              Option Two
            </option>,
          ],
          "className": "select__field",
          "disabled": false,
          "name": "email",
          "onChange": [MockFunction],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "one",
            "nodeType": "host",
            "props": Object {
              "children": "Option One",
              "value": "one",
            },
            "ref": null,
            "rendered": "Option One",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "two",
            "nodeType": "host",
            "props": Object {
              "children": "Option Two",
              "value": "two",
            },
            "ref": null,
            "rendered": "Option Two",
            "type": "option",
          },
        ],
        "type": "select",
      },
      undefined,
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className="input__label"
            htmlFor="email"
          >
            Email
          </label>,
          <select
            className="select__field"
            disabled={false}
            name="email"
            onChange={[MockFunction]}
          >
            <option
              value="one"
            >
              Option One
            </option>
            <option
              value="two"
            >
              Option Two
            </option>
          </select>,
          undefined,
          undefined,
        ],
        "className": "input",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Email",
            "className": "input__label",
            "htmlFor": "email",
          },
          "ref": null,
          "rendered": "Email",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                value="one"
              >
                Option One
              </option>,
              <option
                value="two"
              >
                Option Two
              </option>,
            ],
            "className": "select__field",
            "disabled": false,
            "name": "email",
            "onChange": [MockFunction],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "one",
              "nodeType": "host",
              "props": Object {
                "children": "Option One",
                "value": "one",
              },
              "ref": null,
              "rendered": "Option One",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "two",
              "nodeType": "host",
              "props": Object {
                "children": "Option Two",
                "value": "two",
              },
              "ref": null,
              "rendered": "Option Two",
              "type": "option",
            },
          ],
          "type": "select",
        },
        undefined,
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
